lib/whitehall/asset_manager_storage.rb

```diff
@@ -11,14 +11,31 @@ def store!(file)
     FileUtils.cp(original_file, temporary_location)
     legacy_url_path = ::File.join("/government/uploads", uploader.store_path)
     draft = uploader.assets_protected?
+
     if uploader.model.respond_to?(:attachable)
-      model_class = uploader.model.attachable && uploader.model.attachable.class.to_s
-      model_id = uploader.model.attachable && uploader.model.attachable.id
+      attachable_model_class = uploader.model.attachable && uploader.model.attachable.class.to_s
+      attachable_model_id = uploader.model.attachable && uploader.model.attachable.id
     end
-    AssetManagerCreateWhitehallAssetWorker.perform_async(temporary_location, legacy_url_path, draft, model_class, model_id)
+
+    AssetManagerCreateWhitehallAssetWorker.perform_async(temporary_location, legacy_url_path, draft, attachable_model_class, attachable_model_id, auth_bypass_ids)
     File.new(uploader.store_path)
   end
 
+  def auth_bypass_ids
+    if uploader.model.respond_to?(:images)
+      uploader.model.images
+         .filter { |image| image.respond_to?(:edition) }
+         .filter_map { |image| image.edition.auth_bypass_id if image.edition.pre_publication? }
+         .uniq
+    elsif uploader.model.respond_to?(:attachable) && uploader.model.attachable.respond_to?(:auth_bypass_id)
+      [uploader.model.attachable.auth_bypass_id]
+    elsif uploader.model.respond_to?(:consultation_response_form) && !uploader.model.consultation_response_form.nil?
+      [uploader.model.consultation_response_form.consultation_participation.consultation.auth_bypass_id]
+    else
+      []
+    end
+  end
```

Comment:
Refactor:

- Create methods on each model – i.e. `ImageData#auth_bypass_ids`, `AttachmentData#auth_bypass_ids`, etc.
- Each model becomes responsible for knowing how to get bypass IDs given its own associations
- In this `AssetManagerStorage` class, we can now call `uploader.model.auth_bypass_ids` without having to know about how the model finds those bypass IDs. We can lose the `if ... else` statement, and probably even the entire `AssetManagerStorage#auth_bypass_ids` method.

Reply:
👍 this should now be addressed in the below commit